src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+    support_path = lf_root / 'PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(str(addin_path))
+    automation_class = clr.AddReference(str(automation_path))
+    support_class = clr.AddReference(str(support_path))
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String, Int32, Int64, IntPtr, Double
+    from System.IO import FileAccess
+except KeyError as e:
+    logger.error(f"KeyError {e} during import")
+except ImportError as e:
+    logger.error(f"Unable to import packages: {e}")
+
+
+class LightfieldApplicationManager:
+    def __init__(self, visible: bool) -> None:
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def experiment(self) -> Any:
+        return self._experiment
+
+    def __del__(self) -> None:
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        #NOTE: May need to call '.Dispose()' here again as qt3scan may fail to call 'close'
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting: str, value: Any) -> None:
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+                self.experiment.SetValue(setting, value)
+            else:
+                # TODO: might need to add a proper exception
+                logger.error(f'Invalid value: {value} for setting: {setting}.')
+        else:
+            # TODO: might need to add a proper exception
+            logger.error(f'Invalid setting: {setting}.')
+
+    def get(self, setting: Any) -> Any:
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+        if self.experiment.Exists(setting):
+            value = self.experiment.GetValue(setting)
+        else:
+            value = []
+            logger.error(f'Invalid setting: {setting}.')
+        return value
+
+    def load_experiment(self, value: str) -> None:
+        self.experiment.Load(String(value))
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachDate, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachTime, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachIncrement, True)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationIncrementNumber, Int32(0))
```

Comment:
Consider changing these parameters from hard-coded to parameters in a yaml file the user can change

Reply:
@vasilisniaouris there's a larger discussion that needs to happen about how data are stored with the spectrometer. The spectrometer API isn't great. Apparently Max wanted to just get something out the door now and deal with it later. Also, these settings affect how the data are retrieved.

@cmordi -- regarding a conversation we had about this, I no longer think the QT3ScanPrincetonSpectrometerController class should have to care about this. So keep these calls in this class. But I do think, perhaps, these calls should go somewhere else -- possibly the constructor.